File: solace/cmd/deprecated/Stand.java
package solace.cmd.deprecated;

import solace.game.*;
import solace.net.*;

/**
 * Allows the player to stand up from a sitting, resting, or sleeping state.
 * @author Ryan Sandor Richards
 */
public class Stand extends PlayStateCommand {
  public Stand(solace.game.Character ch) {
    super("stand", ch);
  }

  public void run(Connection c, String []params) {
    Room room = character.getRoom();

    if (character.isStandingOrFighting()) {
      character.sendln("You are already standing.");
      return;
    }

    String characterMessage = "";
    String roomFormat = "";

    if (character.isSitting()) {
      characterMessage = "You stand up.";
      roomFormat = "%s stands up.";
    } else if (character.isSleeping()) {
      characterMessage = "You awake and stand up.";
      roomFormat = "%s wakes and stands up.";
    } else if (character.isResting()) {
      characterMessage = "You stop resting and stand up.";
      roomFormat = "%s stops resting and stands up.";
    }

    room.sendMessage(
      String.format(roomFormat, character.getName()),
      character
    );
    character.sendln(characterMessage);
    character.setStanding();
  }
}

File: script/commands/state/stand.js
'use strict';

/**
 * Allows the player to stand up from a sitting, resting, or sleeping state.
 * @author Ryan Sandor Richards
 */
Commands.add('stand', {
  aliases: ['wake'],
  run: function (player, params) {
    if (player.isStandingOrFighting()) {
      player.sendln("You are already standing.");
      return;
    }

    var playerMessage = "";
    var roomFormat = "";

    if (player.isSitting()) {
      playerMessage = "You stand up.";
      roomFormat = "%s stands up.";
    } else if (player.isSleeping()) {
      playerMessage = "You awake and stand up.";
      roomFormat = "%s wakes and stands up.";
    } else if (player.isResting()) {
      playerMessage = "You stop resting and stand up.";
      roomFormat = "%s stops resting and stands up.";
    }

    player.getRoom().sendMessage(
      format(roomFormat, player.getName()),
      player
    );
    player.sendln(playerMessage);
    player.setStanding();
  }
});
